### PR TITLE
fix: do not lower address for non-EVM accounts during account creation

### PR DESCRIPTION
**Pull request by ccharly**
Some non-EVM account's addresses might be case-sensitive like Solana.

We didn't have any issue with Bitcoin native segwit addresses (since they are lower-case already), but to avoid introducing this "lower-case" behavior everywhere we will only modify the address for EVM accounts.

cc: @aganglada

### CODE REVIEW

**Comment by ccharly** [comment]
@metamaskbot publish-previews

File: packages/keyring-snap-bridge/src/SnapKeyring.ts [hunk 1]
```diff
@@ -100,6 +101,21 @@ export type SnapKeyringCallbacks = {
...
+/**
+ * Normalize account's address.
+ *
+ * @param account - The account.
+ * @returns The normalized account's address.
+ */
+function normalizeAccountAddress(account: KeyringAccount): string {
+  // FIXME: Is it required to lowercase the address here? For now we'll keep this behavior
+  // only for Ethereum address and use the original one for other non-EVM accounts.
+  // For example, solana addresses are case-sensitives.
```
Comment (danroc) [inline packages/keyring-snap-bridge/src/SnapKeyring.ts:113]:
```suggestion
  // For example, Solana addresses are case-sensitives.
```

File: packages/keyring-snap-bridge/src/SnapKeyring.ts [hunk 2]
```diff
@@ -100,6 +101,21 @@ export type SnapKeyringCallbacks = {
...
 
+/**
+ * Normalize account's address.
+ *
+ * @param account - The account.
+ * @returns The normalized account's address.
+ */
+function normalizeAccountAddress(account: KeyringAccount): string {
+  // FIXME: Is it required to lowercase the address here? For now we'll keep this behavior
+  // only for Ethereum address and use the original one for other non-EVM accounts.
```
Comment (danroc) [inline packages/keyring-snap-bridge/src/SnapKeyring.ts:112]:
```suggestion
  // only for Ethereum addresses and use the original one for other non-EVM accounts.
```

**Review by gantunesr — APPROVED** [review]

**Review by zone-live — APPROVED** [review]

File: packages/keyring-snap-bridge/src/SnapKeyring.test.ts [hunk 3]
```diff
@@ -592,9 +647,13 @@ describe('SnapKeyring', () => {
...
       const addresses = await keyring.getAccounts();
-      expect(addresses).toStrictEqual(
-        accounts.map((a) => a.address.toLowerCase()),
-      );
+      expect(addresses).toStrictEqual([
+        ethEoaAccount1.address.toLowerCase(),
+        ethEoaAccount2.address.toLowerCase(),
+        ethEoaAccount3.address.toLowerCase(),
+        ethErc4337Account.address.toLowerCase(),
+        btcP2wpkhAccount.address,
```
Comment (danroc) [inline packages/keyring-snap-bridge/src/SnapKeyring.test.ts:655]: Can be handled in a different PR, but we should have a Solana account too.

**Pull request merged**